File: mobile/src/com/mycompany/myapp/entities/Commande.java
package com.mycompany.myapp.entities;

import java.util.Date;

public class Commande {
    private int id_commande;
    private Date dateCommande ;
    private String nomClient ;
    private String mailClient ;
    private String adresseLivraison;
    private Float prixTotal ;
    private String status ;
    private Panier panier;

    
     public Commande() {
        
    }
    public Commande(int id_commande, Date dateCommande, String nomClient, String mailClient, String adresseLivraison, Float prixTotal, String status) {
        this.id_commande = id_commande;
        this.dateCommande = dateCommande;
        this.nomClient = nomClient;
        this.mailClient = mailClient;
        this.adresseLivraison = adresseLivraison;
        this.prixTotal = prixTotal;
        this.status = status;
    }

    public int getId_commande() {
        return id_commande;
    }

    public void setId_commande(int id_commande) {
        this.id_commande = id_commande;
    }

    public Date getDateCommande() {
        return dateCommande;
    }

    public void setDateCommande(Date dateCommande) {
        this.dateCommande = dateCommande;
    }

    public String getMailClient() {
        return mailClient;
    }

    public void setMailClient(String mailClient) {
        this.mailClient = mailClient;
    }

    public String getAdresseLivraison() {
        return adresseLivraison;
    }

    public void setAdresseLivraison(String adresseLivraison) {
        this.adresseLivraison = adresseLivraison;
    }

    public Float getPrixTotal() {
        return prixTotal;
    }

    public void setPrixTotal(Float prixTotal) {
        this.prixTotal = prixTotal;
    }
    public void setNomClient(String nomClient) {
        this.nomClient = nomClient;
    }

    public String getNomClient() {
        return this.nomClient;
    }
        
    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Panier getPanier() {
        return panier;
    }

    public void setPanier(Panier panier) {
        this.panier = panier;
    }
    
    
     @Override
  public String toString() {
        return "Commande: {" + "id_commande : " + id_commande + ", Date: " + dateCommande + ", Nom de Client : " + nomClient + ", adresse mail de Client : " + mailClient +
     ", adresse de Livraison: " + adresseLivraison + ", Prix Total: " + prixTotal + " , Status :" + " , Panier :"  + '}';
    }   

}

File: web/src/Entity/Commande.php
<?php

namespace App\Entity;

use App\Repository\CommandeRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use phpDocumentor\Reflection\Types\Integer;
use Symfony\Component\Validator\Constraints as Assert;


#[ORM\Entity(repositoryClass: CommandeRepository::class)]
class Commande
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $dateCommande = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"le nom doit etre non vide !")]
    #[Assert\Length( min : 3,minMessage :"Entrer un nom au min de 5 caracteres")]
    private ?string $nomClient = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Entrer votre adresse mail")]
    #[Assert\Email(message : "L'adresse email '{{ value }}' n'est pas valide.")]
    private ?string $mailClient = null;


    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"adresse doit etre non vide !")]
    #[Assert\Length( min : 10,minMessage :"Entrer une adresse au min de 10 caracteres")]
    private ?string $adresseLivraison = null;

    #[ORM\Column]
    private ?float $prixTotal = null;

    #[ORM\Column(length: 255)]
    private ?string $status = null;

    #[ORM\OneToOne(cascade: ['persist', 'remove'])]
    private ?Panier $panier = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDateCommande(): ?\DateTimeInterface
    {
        return $this->dateCommande;
    }

    public function setDateCommande(\DateTimeInterface $dateCommande): self
    {
        $this->dateCommande = $dateCommande;

        return $this;
    }

    public function getNomClient(): ?string
    {
        return $this->nomClient;
    }

    public function setNomClient(string $nomClient): self
    {
        $this->nomClient = $nomClient;

        return $this;
    }


    public function getmailClient(): ?string
    {
        return $this->mailClient;
    }

    public function setMailClient(string $mailClient): self
    {
        $this->mailClient = $mailClient;

        return $this;
    }


    public function getAdresseLivraison(): ?string
    {
        return $this->adresseLivraison;
    }

    public function setAdresseLivraison(string $adresseLivraison): self
    {
        $this->adresseLivraison = $adresseLivraison;

        return $this;
    }

    public function getPrixTotal(): ?float
    {
        return $this->prixTotal;
    }

    public function setPrixTotal(float $prixTotal): self
    {
        $this->prixTotal = $prixTotal;

        return $this;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(string $status): self
    {
        $this->status = $status;

        return $this;
    }

    public function getPanier(): ?panier
    {
        return $this->panier;
    }

    public function setPanier(?panier $panier): self
    {
        $this->panier = $panier;

        return $this;
    }
}
